File: graph/src/main/groovy/net/twisterrob/gradle/graph/vis/d3/javafx/GraphWindow.java
package net.twisterrob.gradle.graph.vis.d3.javafx;

import org.gradle.api.initialization.Settings;

import javafx.application.Platform;
import javafx.stage.Stage;

public class GraphWindow extends net.twisterrob.gradle.graph.vis.d3.GraphWindow {
	private final Stage window;

	public GraphWindow(Stage stage) {
		stage.setScene(createScene(stage.getWidth(), stage.getHeight()));
		this.window = stage;

		window.setTitle("Gradle Build Graph");
	}

	public Stage getStage() {
		return window;
	}

	@Override public void showUI(Settings project) {
		super.showUI(project);
		window.setTitle(String.format("%s - Gradle Build Graph", project.getRootProject().getName()));
		Platform.runLater(new Runnable() {
			@Override public void run() {
				window.show(); // delay display so that bridge's runLater runs first
			}
		});
	}

	@Override public void closeUI() {
		super.closeUI();
		Platform.runLater(new Runnable() {
			@Override public void run() {
				window.close();
			}
		});
	}
}

File: graph/src/main/kotlin/net/twisterrob/gradle/graph/vis/d3/javafx/GraphWindow.kt
package net.twisterrob.gradle.graph.vis.d3.javafx

import javafx.application.Platform
import javafx.stage.Stage
import net.twisterrob.gradle.graph.vis.d3.GraphWindow
import org.gradle.api.initialization.Settings

class GraphWindow(
	val stage: Stage
) : GraphWindow() {

	init {
		stage.scene = createScene(stage.width, stage.height)
		stage.title = "Gradle Build Graph"
	}

	override fun showUI(project: Settings) {
		super.showUI(project)
		stage.title = "${project.rootProject.name} - Gradle Build Graph"
		// Delay display so that bridge's runLater runs first.
		Platform.runLater(stage::show)
	}

	override fun closeUI() {
		super.closeUI()
		Platform.runLater(stage::close)
	}
}
